WordPress/gutenberg
# Avoid adding default block to empty widget areas in customizer

## Description

Fix #32968.

The block editor will inject a default block whenever the content is empty, preventing it from losing focus. This doesn't play nice in the Widgets Customizer though, as an empty block will still create a widget and thus pollute the sidebar. The standalone widgets screen doesn't suffer from this because it will never be emptied, there's always going to be Widget Area blocks.

The solution is to unregister the default block in the customizer, so that the editor won't try to insert a default block whenever the editor is empty. This has a side-effect though as now the focus will be lost after deleting the last block. A possible solution to that would be to manually move the focus to the button block appender whenever that happens.

Another solution to this problem would be to skip creating the empty widget when there's only one block and which is an empty default block. This is more semantically correct and we won't lose the focus. But it might be confusing to the users though 🤔. 

I'm open to any other suggestions.

## How has this been tested?



1. Go to Appearance -> Customize -> Widgets
2. Try to empty a widget area by deleting all blocks
3. Hit "Publish"
4. Confirm that all widgets are deleted in the frontend

## Screenshots 

https://user-images.githubusercontent.com/7753001/123385317-fb85f300-d5c7-11eb-92fa-f8c2161b880d.mp4


## Types of changes




Bug fix

## Checklist:
